### ISSUE TITLE
unexpected token: CREATE

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#288]
Version 2.0.2

I tried to create a reproducible example of #134 
I stripped down the two class files to the minimum, but i get a different error somehow.
Maybe this is related.

Can be reproduced with the attached classes.
[Source.zip](https://github.com/Riverside-Software/sonar-openedge/files/1088830/Source.zip)

**Comment by gquerret** [comment]
Which message do you get ? I don't get any error under 2.0.2.

**Comment by movedoa** [comment]
```
[sonar:sonar] Error during code parsing for Web/CustomerModelTest.cls
[sonar:sonar] org.prorefactor.refactor.RefactorException: C:\Buildtools\Jenkins\workspace\Sonar_Runner\Web\CustomerModelTest.cls:11:7: unexpected token: CREATE
[sonar:sonar] 	at org.prorefactor.treeparser.ParseUnit.parse(ParseUnit.java:152)
[sonar:sonar] 	at org.prorefactor.treeparser.ParseUnit.treeParser01(ParseUnit.java:179)
[sonar:sonar] 	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.parseMainFile(OpenEdgeProparseSensor.java:185)
[sonar:sonar] 	at org.sonar.plugins.openedge.sensor.OpenEdgeProparseSensor.execute(OpenEdgeProparseSensor.java:136)
[sonar:sonar] 	at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
[sonar:sonar] 	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:57)
[sonar:sonar] 	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:49)
[sonar:sonar] 	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
[sonar:sonar] 	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:175)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:262)
[sonar:sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:257)
[sonar:sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:247)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
[sonar:sonar] 	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
[sonar:sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
[sonar:sonar] 	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:118)
[sonar:sonar] 	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:117)
[sonar:sonar] 	at org.sonar.runner.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:67)
[sonar:sonar] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[sonar:sonar] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[sonar:sonar] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 	at java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 	at org.sonar.runner.impl.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:61)
[sonar:sonar] 	at com.sun.proxy.$Proxy2.execute(Unknown Source)
[sonar:sonar] 	at org.sonar.runner.api.EmbeddedRunner.doExecute(EmbeddedRunner.java:274)
[sonar:sonar] 	at org.sonar.runner.api.EmbeddedRunner.runAnalysis(EmbeddedRunner.java:165)
[sonar:sonar] 	at org.sonar.runner.api.EmbeddedRunner.runAnalysis(EmbeddedRunner.java:152)
[sonar:sonar] 	at org.sonarsource.scanner.ant.SonarQubeTask.launchAnalysis(SonarQubeTask.java:72)
[sonar:sonar] 	at org.sonarsource.scanner.ant.SonarQubeTask.execute(SonarQubeTask.java:55)
[sonar:sonar] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
[sonar:sonar] 	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
[sonar:sonar] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 	at java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[sonar:sonar] 	at org.apache.tools.ant.Task.perform(Task.java:348)
[sonar:sonar] 	at org.apache.tools.ant.Target.execute(Target.java:435)
[sonar:sonar] 	at org.apache.tools.ant.Target.performTasks(Target.java:456)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
[sonar:sonar] 	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
[sonar:sonar] 	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:440)
[sonar:sonar] 	at org.apache.tools.ant.taskdefs.CallTarget.execute(CallTarget.java:105)
[sonar:sonar] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
[sonar:sonar] 	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
[sonar:sonar] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[sonar:sonar] 	at java.lang.reflect.Method.invoke(Method.java:498)
[sonar:sonar] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
[sonar:sonar] 	at org.apache.tools.ant.Task.perform(Task.java:348)
[sonar:sonar] 	at org.apache.tools.ant.Target.execute(Target.java:435)
[sonar:sonar] 	at org.apache.tools.ant.Target.performTasks(Target.java:456)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
[sonar:sonar] 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
[sonar:sonar] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
[sonar:sonar] 	at org.apache.tools.ant.Main.runBuild(Main.java:851)
[sonar:sonar] 	at org.apache.tools.ant.Main.startAnt(Main.java:235)
[sonar:sonar] 	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
[sonar:sonar] 	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
[sonar:sonar] Caused by: antlr.NoViableAltException: unexpected token: CREATE
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.createstatement(ProParser.java:6619)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.statement(ProParser.java:2101)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.blockorstate(ProParser.java:422)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.code_block(ProParser.java:453)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.methodstate(ProParser.java:10389)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.statement(ProParser.java:2332)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.blockorstate(ProParser.java:422)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.code_block(ProParser.java:453)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.classstate(ProParser.java:5681)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.statement(ProParser.java:2038)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.blockorstate(ProParser.java:422)
[sonar:sonar] 	at org.prorefactor.proparse.ProParser.program(ProParser.java:154)
[sonar:sonar] 	at org.prorefactor.treeparser.ParseUnit.parse(ParseUnit.java:141)
[sonar:sonar] 	... 62 common frames omitted
[sonar:sonar] 2 files proparse'd, 0 XML files, 1 failure(s)
```

**Comment by gquerret** [comment]
You're right, I've tested with 2.0.3-SNAPSHOT, not with 2.0.2.
As 2.0.3 contains the fix for #284, this changed again the way `CREATE` statements are handled. This message doesn't appear again.
2.0.3 will also include b1574cc8cf531d00b4244a03afc59871b5e7a4d5, and starting 2.0.4 it will be possible to have easy access to properties / TT / methods from any class in the hierarchy.
So I'll close this issue in the mean time, and I'll let you comment on #134 when 2.0.4 is ready.
